### PR TITLE
UpgradeManager expose functions

### PR DESCRIPTION
**Pull request by luandro**
Solves #536

### CODE REVIEW

File: src/backend/lib/upgrade-download.js [hunk 1]
```diff
@@ -178,9 +190,8 @@ class Search extends EventEmitter {
     if (upgrade.arch.indexOf(this.storage.getLocalArch()) === -1) return false;
     if (upgrade.platform !== this.storage.getLocalPlatform()) return false;
     if (!semver.valid(upgrade.version)) return false;
-    if (!semver.gt(upgrade.version, this.storage.getLocalVersion()))
+    if (semver.lt(upgrade.version, this.storage.getLocalVersion()))
       return false;
-    if (this.context.upgrades.some(u => u.hash === upgrade.hash)) return false;
```
Comment (gmaclennan) [inline src/backend/lib/upgrade-download.js:183]: I think this needs to be rebased, this shouldn't be in here right?

File: src/backend/lib/upgrade-download.js [hunk 2]
```diff
@@ -124,16 +124,28 @@ class Search extends EventEmitter {
...
+                });
+
+                const newUpgrades = filtered;
+                const newUpgrade = Object.assign(upgrade, { host, port });
+                newUpgrades.push(newUpgrade);
+                searchLog("new upgrade candidate found", newUpgrade);
+                const newContext = Object.assign(this.context, {
+                  upgrades: newUpgrades,
+                });
+                this.setState(UpgradeState.Search.Searching, newContext);
```
Comment (gmaclennan) [inline src/backend/lib/upgrade-download.js:148]: Should this be in here?

File: src/backend/lib/upgrade-manager.js [hunk 3]
```diff
@@ -104,14 +71,45 @@ class UpgradeManager {
...
+
+    if (this.upgradeSearchTimeout) {
+      clearTimeout(this.upgradeSearchTimeout);
+    }
+  }
+
+  getState() {
+    log("got request for state");
+    this.emitState();
+  }
```
Comment (gmaclennan) [inline src/backend/lib/upgrade-manager.js:103]: I think this should return state, not emit it.

**Review by gmaclennan — CHANGES_REQUESTED** [review]
I have reviewed the code, but I have not reviewed the tests. I think there is some code that shouldn't be in here, probably because a rebase is needed?

File: src/backend/lib/upgrade-download.js (hunk 1 above)
Comment (hackergrrl) [reply src/backend/lib/upgrade-download.js:183]: right

**Comment by hackergrrl** [comment]
This is probably at least partly my fault. I did a force push late last week. Now that y'all are working on this branch too I'll refrain from that in the future.

Can you rebase this & I'll review?

**Comment by hackergrrl** [comment]
This should now be superceded by https://github.com/digidem/mapeo-mobile/pull/567

**Pull request closed without merging**